ReactNativeClient/lib/services/KeymapService.ts

```diff
@@ -0,0 +1,317 @@
+		{ accelerator: 'Shift+Cmd+S', command: 'focusSidebar' },
+		{ accelerator: 'Shift+Cmd+L', command: 'focusNoteList' },
+		{ accelerator: 'Shift+Cmd+N', command: 'focusNoteTitle' },
+		{ accelerator: 'Shift+Cmd+B', command: 'focusNoteBody' },
+		{ accelerator: 'Option+Cmd+S', command: 'toggleSidebar' },
+		{ accelerator: 'Cmd+L', command: 'toggleVisiblePanes' },
+		{ accelerator: 'Cmd+0', command: 'zoomActualSize' },
+		{ accelerator: 'Cmd+E', command: 'startExternalEditing' },
+		{ accelerator: 'Option+Cmd+T', command: 'setTags' },
+		{ accelerator: 'Cmd+G', command: 'gotoAnything' },
+		{ accelerator: 'F1', command: 'help' },
+	];
+
+	static readonly defaultKeymap_OtherPlatforms: KeymapItem[] = [
+		{ accelerator: 'Ctrl+N', command: 'newNoteItem' },
+		{ accelerator: 'Ctrl+T', command: 'newTodoItem' },
+		{ accelerator: 'Ctrl+S', command: 'synchronize' },
+		{ accelerator: 'Ctrl+P', command: 'printItem' },
+		{ accelerator: 'Ctrl+Q', command: 'quit' },
+		{ accelerator: 'Ctrl+Alt+I', command: 'insertTemplate' },
+		{ accelerator: 'Ctrl+C', command: 'copy' },
+		{ accelerator: 'Ctrl+X', command: 'cut' },
+		{ accelerator: 'Ctrl+V', command: 'paste' },
+		{ accelerator: 'Ctrl+A', command: 'selectAll' },
+		{ accelerator: 'Ctrl+B', command: 'textBold' },
+		{ accelerator: 'Ctrl+I', command: 'textItalic' },
+		{ accelerator: 'Ctrl+K', command: 'textLink' },
+		{ accelerator: 'Ctrl+`', command: 'textCode' },
+		{ accelerator: 'Ctrl+Shift+T', command: 'insertDateTime' },
+		{ accelerator: 'F6', command: 'focusSearch' },
+		{ accelerator: 'Ctrl+F', command: 'showLocalSearch' },
+		{ accelerator: 'Ctrl+Shift+S', command: 'focusSidebar' },
+		{ accelerator: 'Ctrl+Shift+L', command: 'focusNoteList' },
+		{ accelerator: 'Ctrl+Shift+N', command: 'focusNoteTitle' },
+		{ accelerator: 'Ctrl+Shift+B', command: 'focusNoteBody' },
+		{ accelerator: 'F10', command: 'toggleSidebar' },
+		{ accelerator: 'Ctrl+L', command: 'toggleVisiblePanes' },
+		{ accelerator: 'Ctrl+0', command: 'zoomActualSize' },
+		{ accelerator: 'Ctrl+E', command: 'startExternalEditing' },
+		{ accelerator: 'Ctrl+Alt+T', command: 'setTags' },
+		{ accelerator: 'Ctrl+,', command: 'config' },
+		{ accelerator: 'Ctrl+G', command: 'gotoAnything' },
+		{ accelerator: 'F1', command: 'help' },
+	];
+
+	static readonly defaultKeymap = shim.isMac()
+		? KeymapService.defaultKeymap_macOS
+		: KeymapService.defaultKeymap_OtherPlatforms;
+
+	constructor() {
+		super();
+
+		this.keymap = KeymapService.initKeymap();
+		this.logger_ = new Logger();
+
+		const customKeymapPath = `${Setting.value('profileDir')}/keymap-desktop.json`;
+		if (fs.existsSync(customKeymapPath)) {
+			try {
+				const customKeymap = JSON.parse(fs.readFileSync(customKeymapPath, 'utf-8'));
+				this.setKeymap(customKeymap);
+			} catch (err) {
+				const msg = err.message ? err.message : '';
+				throw new Error(
+					`KeymapService: Failed to override keymap at ${customKeymapPath}!\n\n${msg}`
+				);
+			}
+		}
```

Comment:
What happens in case of an error? Can the app starts? What keyboard shortcuts are available?

Reply:
Errors are shown in a message box. The app will be closed when user presses the OK button.